payroll-processor-client/src/app/employee/employee-list/state/employee-list.service.ts

```diff
@@ -0,0 +1,36 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable, throwError } from 'rxjs';
+import { catchError, map } from 'rxjs/operators';
+
+import { Employee } from './employee-list.model';
+import { EmployeeListStore } from './employee-list.store';
+
+import { EnvService } from 'src/app/shared/env.service';
+
+@Injectable({ providedIn: 'root' })
+export class EmployeeListService {
+  private readonly apiUrl: string;
+  private employees: Employee[];
+
+  constructor(
+    private http: HttpClient,
+    envService: EnvService,
+    private employeeListStore: EmployeeListStore,
+  ) {
+    this.apiUrl = envService.apiRootUrl;
+  }
+
+  getEmployees(): Observable<void | Employee[]> {
+    return this.http.get<Employee[]>(`${this.apiUrl}/Employees`).pipe(
```

Comment:
I think you'll want to call `this.store.setLoading(true)` before calling the API.

Also, we can subscribe here in the service instead of returning an observable to the component.

There's a lot of nuance about subscriptions, but for this scenario where we are making a simple HTTP request, calling `.subscribe()` in the service and pushing the results into the store is a perfectly fine pattern.

Since we'll be using `.subscribe()` here, we can do all our work in the observer we pass to `.subscribe()`

```typescript
.pipe(
    catchError((err) => {
      this.employeeListStore.setError({ message: 'Could not load employees' });
      return of([]);
    })
)
.subscribe({
  next: (employees) => this.employeesListStore.set(employees),
  complete: () => this.employeesListStore.setLoading(false)
});
```

Reply:
Fixed! This is good insight, although i suspect there is still a lot to learn as far as where to subscribe for http requests.

Reply:
Some of this is talked about in [Akita's docs on error handling](https://datorama.github.io/akita/docs/additional/notifications) and on [subscription management](https://netbasal.gitbook.io/akita/other/best-practices#subscription-management).